tracing-tracy/src/lib.rs

```diff
@@ -273,11 +323,11 @@ impl Visit for TracyEventFieldVisitor {
         // FIXME: this is a very crude formatter, but we don’t have
         // an easy way to do anything better...
         if self.first {
-            let _ = write!(&mut self.dest, "{} = {:?}", field.name(), value);
             self.first = false;
         } else {
-            let _ = write!(&mut self.dest, ", {} = {:?}", field.name(), value);
+            self.dest.push_str(", ");
         }
+        let _ = write!(&mut self.dest, "{} = {value:?}", field.name());
```

Comment:
I believe the original approach gives the formatting infrastructure a way to do the same thing with potentially one less (re-)allocation, doesn’t it?

Reply:
Yeah, I can't tell without stepping through the assembly which is better, so I went back to the old version: https://rust.godbolt.org/z/W7Y8vdxzb

I hand-wrote the str and bool implementations since they can be done without the fmt infrastructure and with a single allocation.